rusoto/credential/src/profile.rs

```diff
@@ -171,6 +177,17 @@ impl ProfileProvider {
     {
         self.profile = profile.into();
     }
+
+    /// Returns `true` if this provider instance will not execute any external processes to obtain
+    /// credentials.
+    pub fn is_secure(&self) -> bool {
+        self.secure
+    }
+
+    /// Ensure this provider instance does not call any external processes to obtain credentials.
+    pub fn secure(&mut self) {
+        self.secure = true;
+    }
```

Comment:
Is behavior like this present in any other SDKs? And this is present to disable credential_process by default in rusoto_sts::DefaultCredentialsProvider?

Reply:
To maintain backward compatibility this is an opt-out hatch for users of the `ProfileProvider`, 
and the only place it is called right now is in the new STS [DefaultProviderChain](https://github.com/rusoto/rusoto/pull/1577/files#diff-9ede61153b201e5b29f7a64bfb5d254bR85). From a first glance it looks that for example in 
`Boto3` (Python SDK) one cannot opt-out from using `credential_process`. However, that is a dangerous insecure default, that might make some sense for AWS CLI, but which we should not necessarily adopt for rusoto. I think that rusoto should follow a "secure by default" approach and that such potential insecure behaviour/features must be turned on (opted-in) explicitly. 

Recommended further readings:
- https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html
- https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#sourcing-credentials-from-external-processes